### PR TITLE
refactor(components): remove addons-bluemix components

### PR DESCRIPTION
**Pull request by joshblack**
Depends on: carbon-design-system/carbon-addons-bluemix#3

Closes: Bluemix/carbon-issues#364 (internal)

`carbon-components` contains deprecation notices for the following components: 

- Cards
- Detail Page Header
- Interior Left Nav
- Modules
- Order Summary

#### Changelog

**New**

**Changed**

- Deprecation notices for the following:
  - Cards
  - Detail Page Header
  - Interior Left Nav
  - Modules
  - Order Summary

**Removed**

### CODE REVIEW

**Comment by joshblack** [comment]
A note for reviewers, would it be better to just add in a deprecation notice instead of removing these before the v8 release?

**Comment by asudoh** [comment]
@joshblack, Thank you for doing this! Please make sure `localhost:3000` with `npm run dev` still work - You may need (circular) dependency to the addons repo.

**Comment by asudoh** [comment]
Oh BTW I'd vote for deprecation notice, e.g. with `console.warn()` - People have complained against breaking changes with minor release update. Thanks!

**Comment by joshblack** [comment]
Cool! Is the `warning` module brought in already? Would love to use that. To follow-up, for deprecation notices what's standard for a SCSS module and a JS module? i.e. where should I put them, what kind of message format should it have, etc.

**Comment by tw15egan** [comment]
Yeah, we'll probably need to add in a deprecated warning until our next breaking release. Are we going to bundle this in with V8? 

As far as standards, I don't think we've formalized anything like that. We have a few mixins that have been deprecated that just tell you that it will eventually be removed. Open to suggestions in this area

**Comment by joshblack** [comment]
I think for the JS-based files, we could add the warning in the constructor so that way it corresponds with when someone uses it (versus it just being required), for the scss stuff I'm unsure how to do that since it seems like it would just give a warning for everyone regardless of if they're actually using the component (unless they are manually specifying their scss imports).

Regardless, I'll update the PR to not remove the files and will start adding in the deprecation notices for the JS-based component files. I'll try and borrow patterns from how React communicates deprecations.

**Comment by joshblack** [comment]
For the `deprecate` mixin, am just logging out:

```scss
WARNING: Deprecated code was found, it should be removed before the next release of Carbon.
REASON: Accessing the `card` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
```

Usage looks like:

```scss
@include exports('card') {
  @include deprecate(
    'Accessing the `card` component from the `carbon-components` package is ' +
    'deprecated. Use the `carbon-addons-bluemix` package instead.'
  ) {
    .bx--card {
      @include reset;
      @include helvetica;
      @include layer('raised');
      // ...
    }
  }
}
```

File: src/components/card/card.js
Comment (asudoh) [inline src/components/card/card.js]: @joshblack Looks that you are experimenting `warning` NPM module. Is it correct? If so - Please ping me if you finished with this - Thanks!

Comment (joshblack) [reply src/components/card/card.js]: Hi @asudoh! Yeah was planning on tackling that today, although it seems like there isn't a "DEV" mode in carbon-components right now to communicate these kinds of warnings. Is that correct? Would hate to include `warning` or `console.error` or something and have it ship to prod.

**Comment by joshblack** [comment]
Just pushed up a commit that helps handle the `__DEV__` expression calls like we were talking about 😄 

Now, when using the JS modules in `development`, the following should be logged:

screen shot 2017-10-25 at 11 36 24 am

In addition, when compiling the CSS (from the sass entrypoint) we get:

```
WARNING: Deprecated code was found, this code will be removed before the next release of Carbon.
REASON: Accessing the `card` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `detail-page-header` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `module` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `interior-left-nav` component from the `carbon-components`package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `order-summary` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
```

File: src/components/modal/_modal.scss
Comment (asudoh) [inline src/components/modal/_modal.scss]: @joshblack Did you really mean to make changes here?

Comment (joshblack) [reply src/components/modal/_modal.scss]: Nope! Thanks for catching 😄

**Review by tw15egan — APPROVED** [review]
Looks good to me! Modules are warning as expected

**Comment by joshblack** [comment]
Figured I'd ask while we wait for another review, did we want to ship this in a minor or major release?

**Comment by tw15egan** [comment]
We'll need to do a major bump for 8, should this one be it? (Even though it's not breaking anything)

**Comment by marijohannessen** [comment]
👍 ✅ 

I think this should be a major release, maybe in the same one you can switch `css--plex` to `true`? I'm going to release the new website tonight around 10pm, and I need the v8 version of the components before that so do you mind getting the commit message ready for a major release so I can merge this around 9:30pm tonight? Thanks!

**Review by asudoh — APPROVED** [review]
Thank you for all the effort @joshblack put here! LGTM assuming that this goes only to our major release that @marijohannessen mentioned.

**Comment by marijohannessen** [comment]
Opened this PR up in a new PR with the Plex changes

**Pull request closed without merging**
